core/src/main/java/com/homeaway/streamplatform/streamregistry/db/dao/impl/ClusterDaoImpl.java

```diff
@@ -0,0 +1,142 @@
+     * @throws ClusterNotFoundException thrown if no cluster is found in the region
+     */
+    public com.homeaway.digitalplatform.streamregistry.ClusterValue getCluster(String vpc, String env, String hint, String actorType) throws ClusterNotFoundException {
+        log.info("Cluster Details: vpc: {}, env: {}, hint: {}, actorType: {}", vpc, env, hint, actorType);
+        com.homeaway.digitalplatform.streamregistry.ClusterKey clusterKey = new com.homeaway.digitalplatform.streamregistry.ClusterKey(vpc, env, hint, null);
+
+        Optional<com.homeaway.digitalplatform.streamregistry.ClusterValue> clusterValue = infraManager.getClusterByKey(clusterKey);
+
+        if (clusterValue.isPresent()) {
+            log.info("Cluster Information found - {}", clusterValue);
+            return new com.homeaway.digitalplatform.streamregistry.ClusterValue(clusterValue.get().getClusterProperties());
+        }
+        // If no cluster information is found set the actorType and look again.
+        clusterKey.setType(actorType);
+
+        clusterValue = infraManager.getClusterByKey(clusterKey);
+
+        if (clusterValue.isPresent()) {
+            log.info("Cluster Information found - {}", clusterValue);
+            return new com.homeaway.digitalplatform.streamregistry.ClusterValue(clusterValue.get().getClusterProperties());
+        } else {
+            log.info("Cluster Information not found for key - {}", clusterKey);
+            throw new ClusterNotFoundException(clusterKey.toString());
+        }
+    }
+
+    /**
+     * Get a Cluster using a clusterName
+     *
+     * @param clusterName - Name of the Cluster from Infra Manager
+     * @return JsonClusterValue - Contains cluster details
+     */
+    public Optional<ClusterValue> getCluster(String clusterName) {
+        log.info("getting cluster detail for cluster - {}", clusterName);
+        Map<com.homeaway.digitalplatform.streamregistry.ClusterKey, com.homeaway.digitalplatform.streamregistry.ClusterValue> allClusters = infraManager.getAllClusters();
+        Map<String, com.homeaway.digitalplatform.streamregistry.ClusterValue> clustersByName = new HashMap<>();
+
+        allClusters.forEach((com.homeaway.digitalplatform.streamregistry.ClusterKey clusterKey, com.homeaway.digitalplatform.streamregistry.ClusterValue clusterValue) -> clustersByName.put(clusterValue.getClusterProperties().get(CLUSTER_NAME), clusterValue));
+
+        log.info("Cluster Info for clusterName - {} is: {}", clusterName, clustersByName.get(clusterName));
+
+        if (clustersByName.get(clusterName) == null) {
+            return Optional.empty();
+        }
+
+        return Optional.of(AvroToJsonDTO.getJsonClusterValue(clustersByName.get(clusterName)));
+    }
+
+    @Override
+    public void upsertCluster(JsonCluster jsonCluster) throws InvalidClusterException {
+        log.info("Upserting Cluster {}", jsonCluster);
+
+        ClusterValidator.isValid(jsonCluster);
```

Comment:
Validating the cluster object and throwing InvalidCluster Exception before upsert